### PR TITLE
Add clk count and blocking statistics to metadata of blocking file

### LINKED ISSUE #85: Add number of encodings in blocking metadata
Before computing similarity score computation and matching, we need to check if the number of encodings in blocking data is consistent with number of encodings in CLK data.

Currently we either load the whole JSON just to get the number of encodings or use ijson to iteratively count the number of encodings.

It would be better to store the count in the metadata and just read this metadata when needing the count.

### PR DESCRIPTION
**Pull request by JoyceXinyueWang**
Close #85

### CODE REVIEW

File: anonlinkclient/utils.py [hunk 1]
```diff
@@ -124,6 +124,11 @@ def generate_candidate_blocks_from_csv(input_f: TextIO,
 
     # step3 - get config meta data
     result['config'] = blocking_config
+
+    # step4 - add CLK counts and blocking statistics to metadata
+    result['clk_count'] = len(encoding_to_blocks_map)
```
Comment (wilko77) [inline anonlinkclient/utils.py:129]: just putting 'clk_count' in the top level of the output is ambiguous. Which CLKs are counted here?

File: tests/test_utils.py [hunk 2]
```diff
@@ -38,6 +38,14 @@ def test_generate_candidate_blocks_from_csv_exception(self):
         with self.assertRaises(ValueError):
             generate_candidate_blocks_from_csv(input_f, input_f, True)
 
+    def test_generate_candidate_blocks_from_csv(self):
```
Comment (wilko77) [inline tests/test_utils.py:41]: that's not a good name for the function, as you don't actually test if blocks where generated and/or correct. 
What you do test is the metadata. Why not call it `test_metadata_in_result` or something like that.

File: anonlinkclient/utils.py (hunk 1 above)
Comment (JoyceXinyueWang) [reply anonlinkclient/utils.py]: CLKs that belongs to any block

File: tests/test_utils.py (hunk 2 above)
Comment (JoyceXinyueWang) [reply tests/test_utils.py]: done

**Pull request merged**